### DISCUSSION TITLE
changes in Valve.vue

### DISCUSSION THREAD

**Discussion by rafaelpsousa** [SCADA-LTS/Scada-LTS#3016] [Ideas]
I don't know if it's a bug but the original code doesn't work for me and I need to make this changes

```
onPointValueUpdate(value) {
	if(isNaN(value)) {
		let state = value == "true";
		let message = state ? "ON" : "OFF";
		this.changeComponentText(`${this.componentId}_value`, message);
		this.onPointEnabledUpdate(message);
	} else {
		this.changeComponentText(`${this.componentId}_value`, parseFloat(value).toFixed(2));
	}
},

onPointEnabledUpdate(message) {
	if (message == "ON") {
		this.changeComponentColor(`${this.componentId}_background_left`, '#1BB350');
		this.changeComponentColor(`${this.componentId}_background_right`, '#1BB350');
	} else if (message == "OFF") {
		this.changeComponentColor(`${this.componentId}_background_left`, '#B8150E');
		this.changeComponentColor(`${this.componentId}_background_right`, '#B8150E');
	}
},

**Comment by Limraj** [comment]
Hi @rafaelpsousa,
First, you need to provide where this code is, second, it's best to provide a scenario that leads to the problem. 
Best regards.

**Comment by rafaelpsousa** [comment]
Hi Kamil,

OK.
1 - this code is located in `scadalts-ui/src/components/SynopticPanel/slts/Valve.vue` and the valve is displayed in the Synoptic Panel

The original code works with the state value, but this state is only changed when something happens, causing confusion with the valve message (ON/OFF) and the color in the component (red, green). In the default value, the valve is green with the message OFF
What I did was use the message (ON/OFF) to control the color of the valve

**Comment by Limraj** [comment]
Okay I see, it looks like we've already fixed it here:
https://github.com/SCADA-LTS/Scada-LTS/issues/2974

and pr:
https://github.com/SCADA-LTS/Scada-LTS/pull/2979

which will be in the next version 2.8.0.
The branch with the current changes is called release/2.8.0 and the changes in this branch will soon be version 2.8.0.

If you could do a test to see if everything is okay, we would be grateful.
